doc/configuration-basics.md

```diff
@@ -0,0 +1,237 @@
+```
+
+* `instance_with_own_color` has it's own color value set - `my-own-color`;
+* `instance_common` uses color value for group `all` - `common-color`;
+* `instance_blue_1` and `instance_blue_2` use color value specific for
+  group `blue` - `blue-color`;
+* `instance_fake_blue` also belongs to `blue` grop, but it also have it's own
+  color value set - ` yellow-color`;
+* `instance_red_1` use color value specific for
+  group `red` - `red-color`;
+
+## Configuring instances and replica sets
+
+The common cases to use groups with `tarantool.cartridge` role are:
+
+* set connection options for instances deployed on one machine;
+* set replicaset options for instances from one replicaset.
+
+What should you understand to connfigure instances:
+
+* each instance is an Ansible host;
+* the hostname is instance alias that is used by role to identify instance.
+
+### Instances
+
+Imagine that you want to deploy three instances:
+
+* `core-1`
+* `storage-1-leader`
+* `storage-1-replica`
+
+For each instance we should specify these [variables](/doc/variables.md):
+* `cartridge_app_name`
+* `cartridge_cluster_cookie`
+* `cartridge_package_path`
+* `config` with one required parameter - instance advertise URI.
+
+`config` variable is specific for each instance, other variables are common
+and can be placed in `all` group.
+
+Let's begin:
+
+```yaml
+---
+all:
+  vars:
+    cartridge_app_name: myapp
+    cartridge_cluster_cookie: secret-cookie
+    cartridge_package_path: ./myapp-1.0.0-0.rpm
+
+  hosts:
+    core-1:
+      config:
+        advertise_uri: '172.19.0.2:3301'
+
+    storage-1-leader:
+      config:
+        advertise_uri: '172.19.0.2:3302'
+
+    storage-1-replica:
+      config:
+        advertise_uri: '172.19.0.3:3301'
+```
+
+We've just declared three hosts (i.e. nodes that we want to configure),
+set common configuration parameters and parameters specific for each instance.
+
+But how should Ansible understand where each instance should be started?
+
+### Connection options
+
+As you understand looking at advertise URIs instances are placed on two machines:
+
+* `machine_1` (`172.19.0.2`):
+  * `core-1`
+  * `storage-1-leader`
+
+* `machine_2` (`172.19.0.3`):
+  * `storage-1-replica`
+
+For each instance we should describe connection options:
+
+* `ansible_host` - IP or name of the machine where instance should be placed;
+* `ansible_user` - the user Ansible "logs in" as.
+
+We will use `groups` to specify options common for several instances:
+
+```yaml
+all:
+  vars:
+    ...
+  hosts:
+    ...
+  children:
+    machine_1:
+      vars:
+        ansible_host: 172.19.0.2
+        ansible_user: vagrant
+
+      hosts:
+        core-1:
+        storage-1-leader:
+
+    machine_2:
+      vars:
+        ansible_host: 172.19.0.3
+        ansible_user: vagrant
+
+      hosts:
+        storage-1-replica:
+```
+
+### Configuring replicasets
+
+Now we should describe replica sets:
+
+* `core-1` (`vshard-router`, `failover-coordinator`):
```

Comment:
```suggestion
* `core_1_replicaset` (`vshard-router`, `failover-coordinator`):
```

Reply:
`core_1_replicaset` is't a replicaset alias, it's Ansible group name.

Reply:
Then core-1-replicaset)